src/main/webapp/app/shared/button/SaveGeneButton.tsx

```diff
@@ -0,0 +1,41 @@
+import { observer } from 'mobx-react';
+import { componentInject } from '../util/typed-inject';
+import { IRootStore } from 'app/stores';
+import React, { useCallback, useState } from 'react';
+import { Button, ButtonProps } from 'reactstrap';
+
+type ISaveGeneButtonProps = StoreProps & {
+  hugoSymbol?: string;
+} & ButtonProps &
+  Omit<React.HTMLAttributes<HTMLButtonElement>, 'onClick' | 'disabled'>;
+
+function SaveGeneButton({ hugoSymbol, firebaseGeneService, ...buttonProps }: ISaveGeneButtonProps) {
+  const [disabled, setDisabled] = useState(false);
+  const onClickHandler = useCallback(async () => {
+    setDisabled(true);
+    try {
+      const isGermline = false;
+      if (hugoSymbol === undefined) {
+        await firebaseGeneService.saveAllGenes(isGermline);
+      } else {
+        await firebaseGeneService.saveGene(isGermline, hugoSymbol);
+      }
+    } catch (e) {
+      console.error(e);
```

Comment:
We should also let user know the save failed.

Reply:
I didn't think it matter since it was for developers, but I think you're right. I'll add a toast message.